engine/graphics/src/null/graphics_null.cpp

```diff
@@ -735,217 +735,293 @@ namespace dmGraphics
         return g_DrawCount;
     }
 
-    struct ShaderBinding
```

Comment:
There's lots of changes in the null adapter, but mostly it is related to using the same code path and structures as the other adapters in an effort to move them closer together. This means we can test the binding code paths better, as even the null adapter can _fully_ support the type reflection. We wouldn't be able to test stuff like nested structs, or more advanced types otherwise.